src/lexer.d

```diff
@@ -88,10 +89,10 @@ version (unittest)
 
 /***********************************************************
  */
-extern (C++) class Lexer
+class Lexer
```

Comment:
To avoid mismatch between C++ declarations and D definitions, you would have to remove `class Lexer` declaration in `lexer.h`